Plugins~/Src/MeshSyncClientBlender/msblenCurveHandler.cpp

```diff
@@ -0,0 +1,119 @@
+#include "pch.h"
+#include "msblenBinder.h"
+#include "msblenCurveHandler.h"
+#include "msblenUtils.h"
+
+namespace bl = blender;
+
+ms::CurvePtr msblenCurveHandler::exportCurve(msblenContextState& state,
+	msblenContextPathProvider& paths,
+	BlenderSyncSettings& settings,
+	const Object* src,
+	MeshSyncClient::AsyncTasksController& asyncTasksController)
+{
+	std::shared_ptr<ms::Curve> ret = ms::Curve::create();
+	
+	Curve* data = (Curve*)src->data;
+
+	ms::Curve& dst = *ret;
+	dst.path = paths.get_path(src);
+	
+	// transform
+	extractTransformData(settings, src, dst);
+
+	if (settings.sync_curves) {
+		auto task = [this, ret, src, data, &state, &settings]() {
+			auto& dst = *ret;
+			doExtractCurveData(state, settings, dst, src, data, dst.world_matrix);
+			state.manager.add(ret);
+		};
+
+		if (settings.multithreaded)
+			asyncTasksController.AddTask(std::launch::async, task);
+		else
+			task();
+	}
+
+	return ret;
+}
+
+void msblenCurveHandler::doExtractCurveData(msblenContextState& state, BlenderSyncSettings& settings, ms::Curve& dst, const Object* obj, Curve* data, mu::float4x4 world)
+{
+	if (settings.sync_curves) {
```

Comment:
can you change this to check for the negation and return if true? will help with reducing nesting

Reply:
done